ipam/provider/aws.go

```diff
@@ -0,0 +1,256 @@
+				// removed manually.
+				aws.logger.Log("level", "error", "message", fmt.Sprintf("failed to close & remove tempfile '%s' - it must be removed manually", fName))
+				return
+			}
+			err = os.Remove(fName)
+			if err != nil {
+				// XXX: Tempfile cannot be removed due to error. It must be
+				// removed manually.
+				aws.logger.Log("level", "error", "message", fmt.Sprintf("failed to close & remove tempfile '%s' - it must be removed manually", fName))
+				return
+			}
+		}()
+
+		{
+			c := chartvalues.APIExtensionsAWSConfigE2EConfig{
+				CommonDomain:         aws.chartValuesConfig.CommonDomain,
+				ClusterName:          clusterName,
+				SSHPublicKey:         aws.chartValuesConfig.SSHPublicKey,
+				VersionBundleVersion: aws.chartValuesConfig.VersionBundleVersion,
+
+				AWS: chartvalues.APIExtensionsAWSConfigE2EConfigAWS{
+					Region:            aws.chartValuesConfig.AWSRegion,
+					APIHostedZone:     aws.chartValuesConfig.AWSAPIHostedZoneGuest,
+					IngressHostedZone: aws.chartValuesConfig.AWSIngressHostedZoneGuest,
+					RouteTable0:       clusterName + "_0",
+					RouteTable1:       clusterName + "_1",
+					VPCPeerID:         hostAWSConfig.Spec.AWS.VPC.PeerID,
+				},
+			}
+
+			values, err := chartvalues.NewAPIExtensionsAWSConfigE2E(c)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			_, err = f.WriteString(values)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+
+		err = framework.HelmCmd(fmt.Sprintf("registry install quay.io/giantswarm/apiextensions-aws-config-e2e-chart:stable -- -n %s --values %s", awsConfigDeploymentName(clusterName), f.Name()))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		return nil
+	}
+	b := backoff.NewConstant(framework.ShortMaxWait, framework.ShortMaxInterval)
+	n := func(err error, delay time.Duration) {
+		log.Println("level", "debug", "message", err.Error())
+	}
+
+	err = backoff.RetryNotify(o, b, n)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (aws *AWS) DeleteCluster(clusterName string) error {
+	deploymentName := awsConfigDeploymentName(clusterName)
+	err := framework.HelmCmd(fmt.Sprintf("delete --purge %s", deploymentName))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (aws *AWS) installCertResources(clusterName string) error {
+	{
+		aws.logger.Log("level", "debug", "message", "installing cert resource chart")
+
+		o := func() error {
+			err := framework.HelmCmd(fmt.Sprintf("delete --purge %s", certDeploymentName(clusterName)))
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			cmdStr := fmt.Sprintf("registry install quay.io/giantswarm/apiextensions-cert-config-e2e-chart:stable -- -n %s --set commonDomain=${COMMON_DOMAIN} --set clusterName=%s", certDeploymentName(clusterName), clusterName)
+			err = framework.HelmCmd(cmdStr)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			return nil
+		}
```

Comment:
Can't be `Host.InstallCertResource()` used here?

Reply:
No because it doesn't accept cluster name. I was thinking if that should be refactored, but then again, there is a ticket in SIG Operator to refactor all helm stuff towards client library instead of using command line escape so I think it can be changed at that point.